src/poliastro/bodies.py

```diff
@@ -52,9 +66,68 @@ def _repr_latex_(self):
 
         """
         return self.__str__()
+    def calculate_soi(self):
```

Comment:
This can be a function outside of the `Body` class:

```
def calculate_soi(body, a, parent=None):
    # Parameter `a` is mandatory now
    # `parent` can be None if `body` already has it
    pass
```